core/src/main/kotlin/org/evomaster/core/problem/rest/service/SecurityRest.kt

```diff
@@ -602,6 +602,96 @@ class SecurityRest {
         }
     }
 
+
+    /**
+     * Check if there is a test case with a 403 and another one with a 200 without authentication.
+     * To check this, there must be a resource. It can either be newly created or already exist,
+     * such as during initialization. While the user who created this
+     * resource can access it (200), the other user cannot (403). However, if a 200 status code is
+     * returned when attempting to access the same resource without sending the authorization header,
+     * it indicates that authorization checks are wrongly ignored if no auth info is set.
+     * Example:
+     * POST /resources/ AUTH1 -> 201 (location header: /resources/42/)
+     * GET /resources/42/ AUTH1 -> 200
+     * GET /resources/42/ AUTH2 -> 403
+     * GET /resources/42/ NOAUTH -> 200
+     */
+    private fun handleForgottenAuthentication() {
+        actionDefinitions.forEach { op ->
+            val i403or401 = RestIndividualSelectorUtils.findIndividuals(
+                individualsInSolution,
+                op.verb,
+                op.path,
+                statusGroup = StatusGroup.G_4xx,
+            )
+
+            if (i403or401.isEmpty()) {
+                return@forEach //there is not any protected resource for this path/verb.
+            }
+
+            val i200 = RestIndividualSelectorUtils.findIndividuals(
+                individualsInSolution,
+                op.verb,
+                op.path,
+                statusGroup = StatusGroup.G_2xx
+            ).flatMap { individual ->
+                individual.individual.seeMainExecutableActions()
+                    .filter { it.auth is NoAuth }
```

Comment:
doesn't seem correct. an individual can have several actions, and you could get here actions not related to the `op` target